### ISSUE TITLE
Changing file permissions is not permitted

### ISSUE THREAD

**Issue by BelKed** [JamesIves/github-pages-deploy-action#1092]
## Describe the bug

There's a problem with the latest version (`v4.3.2`):
```
/usr/bin/chmod: changing permissions of '/home/runner/work/…: Operation not permitted
```
By the way, the previous version (`v4.3.1`) works well...

## Reproduction Steps

Use this action with the settings described in the Workflow section.

## Logs
<details>
<summary>
Log
</summary>

```
…
2022-04-25T16:31:37.5416289Z ##[group]Run JamesIves/github-pages-deploy-action@v4.3.2
2022-04-25T16:31:37.5416540Z with:
2022-04-25T16:31:37.5416700Z   branch: gh-pages
2022-04-25T16:31:37.5416890Z   folder: website
2022-04-25T16:31:37.5417060Z   clean: true
2022-04-25T16:31:37.5417344Z   token: ***
2022-04-25T16:31:37.5417499Z   force: true
2022-04-25T16:31:37.5417668Z ##[endgroup]
2022-04-25T16:31:37.7516192Z 
2022-04-25T16:31:37.7517125Z     ╭━━━╮╭╮╭╮╱╭╮╱╱╭╮╱╱╭━━━╮
2022-04-25T16:31:37.7517690Z     ┃╭━╮┣╯╰┫┃╱┃┃╱╱┃┃╱╱┃╭━╮┃
2022-04-25T16:31:37.7518097Z     ┃┃╱╰╋╮╭┫╰━╯┣╮╭┫╰━╮┃╰━╯┣━━┳━━┳━━┳━━╮
2022-04-25T16:31:37.7518512Z     ┃┃╭━╋┫┃┃╭━╮┃┃┃┃╭╮┃┃╭━━┫╭╮┃╭╮┃┃━┫━━┫
2022-04-25T16:31:37.7518923Z     ┃╰┻━┃┃╰┫┃╱┃┃╰╯┃╰╯┃┃┃╱╱┃╭╮┃╰╯┃┃━╋━━┃
2022-04-25T16:31:37.7519320Z     ╰━━━┻┻━┻╯╱╰┻━━┻━━╯╰╯╱╱╰╯╰┻━╮┣━━┻━━╯
2022-04-25T16:31:37.7519688Z     ╱╱╱╱╱╱╱╱╱╱╱╱╱╱╱╱╱╱╱╱╱╱╱╱╱╭━╯┃
2022-04-25T16:31:37.7520072Z     ╱╱╱╱╱╱╱╱╱╱╱╱╱╱╱╱╱╱╱╱╱╱╱╱╱╰━━╯
2022-04-25T16:31:37.7520435Z     ╭━━━╮╱╱╱╱╱╭╮╱╱╱╱╱╱╱╱╭━━━╮╱╱╭╮
2022-04-25T16:31:37.7520890Z     ╰╮╭╮┃╱╱╱╱╱┃┃╱╱╱╱╱╱╱╱┃╭━╮┃╱╭╯╰╮
2022-04-25T16:31:37.7521579Z     ╱┃┃┃┣━━┳━━┫┃╭━━┳╮╱╭╮┃┃╱┃┣━┻╮╭╋┳━━┳━╮
2022-04-25T16:31:37.7521996Z     ╱┃┃┃┃┃━┫╭╮┃┃┃╭╮┃┃╱┃┃┃╰━╯┃╭━┫┃┣┫╭╮┃╭╮╮
2022-04-25T16:31:37.7522396Z     ╭╯╰╯┃┃━┫╰╯┃╰┫╰╯┃╰━╯┃┃╭━╮┃╰━┫╰┫┃╰╯┃┃┃┃
2022-04-25T16:31:37.7522795Z     ╰━━━┻━━┫╭━┻━┻━━┻━╮╭╯╰╯╱╰┻━━┻━┻┻━━┻╯╰╯
2022-04-25T16:31:37.7523155Z     ╱╱╱╱╱╱╱┃┃╱╱╱╱╱╱╭━╯┃
2022-04-25T16:31:37.7523520Z     ╱╱╱╱╱╱╱╰╯╱╱╱╱╱╱╰━━╯
2022-04-25T16:31:37.7523788Z     
2022-04-25T16:31:37.7524038Z 
2022-04-25T16:31:37.7524381Z     💖 Support: https://github.com/sponsors/JamesIves
2022-04-25T16:31:37.7524860Z     📣 Maintained by James Ives: https://jamesiv.es
2022-04-25T16:31:37.7525114Z 
2022-04-25T16:31:37.7525508Z     🚀 Getting Started Guide: https://github.com/JamesIves/github-pages-deploy-action
2022-04-25T16:31:37.7526170Z     ❓ Discussions / Q&A: https://github.com/JamesIves/github-pages-deploy-action/discussions
2022-04-25T16:31:37.7526789Z     🔧 Report a Bug: https://github.com/JamesIves/github-pages-deploy-action/issues
2022-04-25T16:31:37.7527356Z Checking configuration and starting deployment… 🚦
2022-04-25T16:31:37.7527784Z Deploying using Deploy Token… 🔑
2022-04-25T16:31:37.7528124Z Configuring git…
2022-04-25T16:31:37.7615118Z [command]/usr/bin/git config --global --add safe.directory /home/runner/work/test/test
2022-04-25T16:31:37.7693355Z [command]/usr/bin/git config user.name BelKed
2022-04-25T16:31:37.7733672Z [command]/usr/bin/git config user.email 66956532+BelKed@users.noreply.github.com
2022-04-25T16:31:37.7775200Z [command]/usr/bin/git config core.ignorecase false
2022-04-25T16:31:37.7817018Z [command]/usr/bin/git config --local --unset-all http.https://github.com/.extraheader
2022-04-25T16:31:37.7856782Z [command]/usr/bin/git remote rm origin
2022-04-25T16:31:37.7905969Z [command]/usr/bin/git remote add origin ***github.com/BelKed/test.git
2022-04-25T16:31:37.7939726Z Git configured… 🔧
2022-04-25T16:31:37.7944194Z Starting to commit changes…
2022-04-25T16:31:37.7955609Z [command]/usr/bin/git ls-remote --heads ***github.com/BelKed/test.git refs/heads/gh-pages
2022-04-25T16:31:38.2201611Z 9fc1ad5cfff402231c58d4fdf9ab903c0aa03256	refs/heads/gh-pages
2022-04-25T16:31:38.2209184Z Creating worktree…
2022-04-25T16:31:38.2223819Z [command]/usr/bin/git fetch --no-recurse-submodules --depth=1 origin gh-pages
2022-04-25T16:31:41.8363198Z From https://github.com/BelKed/test
2022-04-25T16:31:41.8364187Z  * branch            gh-pages   -> FETCH_HEAD
2022-04-25T16:31:41.8364701Z  * [new branch]      gh-pages   -> origin/gh-pages
2022-04-25T16:31:41.8394615Z [command]/usr/bin/git worktree add --no-checkout --detach github-pages-deploy-action-temp-deployment-folder
2022-04-25T16:31:41.8426118Z Preparing worktree (detached HEAD a14d24d)
2022-04-25T16:31:41.8464675Z [command]/usr/bin/git checkout -B gh-pages origin/gh-pages
2022-04-25T16:31:42.0775878Z Previous HEAD position was a14d24d Use a newer version of the deployment action
2022-04-25T16:31:42.0781902Z Switched to a new branch 'gh-pages'
2022-04-25T16:31:42.0786090Z branch 'gh-pages' set up to track 'origin/gh-pages'.
2022-04-25T16:31:42.0818303Z [command]/usr/bin/chmod -R +rw /home/runner/work/test/test/website
2022-04-25T16:31:42.0845773Z /usr/bin/chmod: changing permissions of '/home/runner/work/test/test/website': Operation not permitted
2022-04-25T16:31:42.0847158Z /usr/bin/chmod: changing permissions of '/home/runner/work/test/test/website/index.html': Operation not permitted
2022-04-25T16:31:42.0847754Z /usr/bin/chmod: changing permissions of '/home/runner/work/test/test/website/2022.html': Operation not permitted
2022-04-25T16:31:42.0848338Z /usr/bin/chmod: changing permissions of '/home/runner/work/test/test/website/2022-01.html': Operation not permitted
2022-04-25T16:31:42.0848939Z /usr/bin/chmod: changing permissions of '/home/runner/work/test/test/website/2022-03.html': Operation not permitted
2022-04-25T16:31:42.0903956Z /usr/bin/chmod: changing permissions of '/home/runner/work/test/test/website/public/theme.css': Operation not permitted
2022-04-25T16:31:42.0909089Z /usr/bin/chmod: changing permissions of '/home/runner/work/test/test/website/public/jquery.min.js': Operation not permitted
2022-04-25T16:31:42.0918281Z /usr/bin/chmod: changing permissions of '/home/runner/work/test/test/website/public/core.css': Operation not permitted
2022-04-25T16:31:42.0920755Z /usr/bin/chmod: changing permissions of '/home/runner/work/test/test/website/public/play.png': Operation not permitted
2022-04-25T16:31:42.1231014Z /usr/bin/chmod: changing permissions of '/home/runner/work/test/test/website/2022-02.html': Operation not permitted
2022-04-25T16:31:42.1232031Z /usr/bin/chmod: changing permissions of '/home/runner/work/test/test/website/2021.html': Operation not permitted
2022-04-25T16:31:42.1232475Z Running post deployment cleanup jobs… 🗑️
2022-04-25T16:31:42.1232858Z [command]/usr/bin/git checkout -B github-pages-deploy-action/qrvqo6yzh
2022-04-25T16:31:42.2094813Z Switched to a new branch 'github-pages-deploy-action/qrvqo6yzh'
2022-04-25T16:31:42.2147430Z [command]/usr/bin/chmod -R +rw github-pages-deploy-action-temp-deployment-folder
2022-04-25T16:31:42.2206449Z [command]/usr/bin/git worktree remove github-pages-deploy-action-temp-deployment-folder --force
2022-04-25T16:31:42.2472871Z ##[error]The deploy step encountered an error: The process '/usr/bin/chmod' failed with exit code 1 ❌
2022-04-25T16:31:42.2484355Z ##[notice]Deployment failed! ❌
…
```

</details>

## Workflow

```yaml
…
      - name: Deploy to GitHub Pages 🚀
        uses: JamesIves/github-pages-deploy-action@v4.3.2
        with:
          branch: gh-pages
          folder: website
          clean: true
…
```

**Comment by JamesIves** [comment]
This should be fixed in `4.3.3`. I am curious what your entire workflow looks like though, are you running this inside a container, or do you have other steps that occur before the deployment?

**Comment by BelKed** [comment]
Thanks for the quick fix :) Now it works as intended...

And, yes, I have some other steps before deploying:
<details>
<summary>
Workflow file
</summary>

```yaml
name: Publish Gallery

on:
  push:
    branches:
      - master

jobs:
  deploy:
    runs-on: ubuntu-latest

    steps:
      - name: Checkout 🛎️
        uses: actions/checkout@v3

      - name: Build Gallery 🔧
        run: docker run -t -v "$(pwd):/work" thumbsupgallery/thumbsup /bin/sh -c "cd /work/ && thumbsup --config config.json"

      - name: Deploy to GitHub Pages 🚀
        uses: JamesIves/github-pages-deploy-action@v4
        with:
          branch: gh-pages
          folder: website
          clean: true
```

</details>
